src/include/migraphx/op/if_op.hpp

```diff
@@ -0,0 +1,32 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_IF_OP_HPP
+#define MIGRAPHX_GUARD_OPERATORS_IF_OP_HPP
+
+#include <array>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/functional.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/module.hpp>
+#include <cmath>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct if_op
+{
+    std::string name() const { return "if"; }
+
+    shape compute_shape(const std::vector<shape>&, std::vector<module_ref> mods) const
+    {
+        auto out_shapes = mods[0]->get_output_shapes();
```

Comment:
We should still check the input shapes that there is 1 argument, and we should check there are two modules passed and they both have the same shape.

Reply:
fixed